### ISSUE TITLE
Deployments in a separate repo from flow code

### ISSUE THREAD

**Issue by benjamincerigo** [anna-geller/prefect-docker-deployment#1]
Hi, 
thanks for the example template repo, super useful. We have a few questions that we thought you might be able to guide us on.

We are planning to structure our project which used Prefect as such:
- a flow-repo; which contains all the task, flow, and general python code. We intend for this repo to be public.
- a deployments-repo; which contains multiple deployments of various configurations of the flows from the flow-repo. This is so we can deploy the same flows, applied to multiple separate (but similar) projects. We want to keep the details of the projects private and as such also this repo. 

As such we have been looking for answers to questions:
- How can we have flows and deployments in separate repos?
- Would it be possible to have deployments that could use a different version of the flow repo, i.e. some deployed flows use older versions of the code from the flow repo?

Our proposed solution to this from the research and looking at this repo would work as such: 
- The flow repo builds a docker image that is tagged with a version. 
- The deployment repo would then use `--infra docker-container --override image=docker_registry/flow_repo:version`. In the deployment repo it would also seem possible to build the `Deployment` in python using `Deployment.build_from_flow` and defining `entrypoint` rather than the `flow` argument ([if here](https://github.com/PrefectHQ/prefect/blob/c2a76aa57fee8326de0883f1ed708878072eeba9/src/prefect/deployments.py#L556)). 
- This way the flow repo would not be a dependency of the deployment repo. As well as this, different deployments could use different versions of the flow repo by using differently tagged images. 

Some feedback on this would be awesome and we have a few follow-up questions also:
- If possible, could you add some example python Deployment code to this repo to show deploying docker images containing flows?
- In the case that we use `stable` as the docker image tag (image=docker_registry/flow_repo:stable) and have the flow repo persist new images of stable. Would it then be required that we rebuild and reapply deployments that use `stable`? Or would these deployments “pick up” the new image automatically? I guess I am asking if the specific image that will be used for the deployment is copied by Prefect into its own storage location for images, or if the agent uses the image at the defined URL? 

Thanks very much!

**Comment by khuyentran1401** [comment]
Hi @benjamincerigo, currently, you can't have flows and deployments in separate repositories. What kind of configurations do you plan to have in your deployment file that can't be shared? Can you put all of the secrets in blocks and just refer to their names in your deployment file?

**Comment by anna-geller** [comment]
Khuyen provided a perfect answer -- I deliberately didn't provide any Python deployment since I strongly believe that deployment build CLI provides a better UX, but if you need more guidance on that, check out: https://discourse.prefect.io/t/can-i-specify-deployments-using-python-code-rather-than-cli/1458

Also, this may answer many of your questions: https://discourse.prefect.io/t/prefect-deployments-faq/1467

**Comment by benjamincerigo** [comment]
@khuyentran1401 and @anna-geller thanks for the responses. I have been able to implement an (experimental) solution for this issue. I have created an [experiment repo](https://github.com/datavaluepeople/prefect_deployment_experiment) that shows two examples:
- running the hello world docker image through Prefect
- running the `dataflowops` image as defined in this repo through a deployment that is defined in the `prefect_deployment_experiment` repo

Any comments or suggestions would be most appreciated and feel free to close this issue.
